src/compiler/corePublic.ts

```diff
@@ -114,25 +114,31 @@ namespace ts {
 
     /* @internal */
     namespace NativeCollections {
-        declare const Map: MapConstructor | undefined;
-        declare const Set: SetConstructor | undefined;
+        declare const self: any;
+
+        const globals = typeof globalThis !== "undefined" ? globalThis
+                  : typeof global !== "undefined" ? global
+                  : typeof self !== "undefined" ? self
+                  : undefined;

```

Comment:
Minor formatting nit:

```suggestion
        const globals = typeof globalThis !== "undefined" ? globalThis :
                  typeof global !== "undefined" ? global :
                  typeof self !== "undefined" ? self :
                  undefined;
```